### PR TITLE
[WIP] Remove keepalive timers from Workers

### PR DESCRIPTION
**Pull request by mudler**

### CODE REVIEW

File: lib/OpenQA/Worker/Commands.pm [hunk 1]
```diff
@@ -148,19 +145,20 @@ sub websocket_commands {
             if ($job && $job->{id}) {
                 $OpenQA::Worker::Common::job = $job;
                 log_debug("Job " . $job->{id} . " scheduled for next cycle");
+                my $send_status = sub {
+                    log_debug("Sending IMMEDIATELY worker status to $host");
```
Comment (okurz) [inline lib/OpenQA/Worker/Commands.pm]: I saw that you just moved this debug line but I wanted to ask you anyway what it means. I don't understand the english sentence here. Could you clarify that line?

File: lib/OpenQA/Worker/Common.pm [hunk 2]
```diff
@@ -336,31 +340,35 @@ sub call_websocket {
...
+                                (status => 'working', job => $job) x !!($job),
+                                (status => 'free') x !!(!$job),
+                            }});
+                };
+
+                my $i = looks_like_number($instance) ? $instance : 1;
+            # Feature scaling - we do not know the instance maximum per worker, we take DEAD_DETECTION_THRESHOLD as max.
+                my $imax = $i > DEAD_DETECTION_THRESHOLD ? $i : DEAD_DETECTION_THRESHOLD;
+                my $worker_status_timer
+                  = int(MIN_STATUS_TIMER + ((($i - 1) * (DEAD_DETECTION_THRESHOLD - MIN_STATUS_TIMER)) / ($imax - 1)));
```
Comment (okurz) [inline lib/OpenQA/Worker/Common.pm]: IMHO line breaks like these should be avoided so … use a temporary variable assignment?

File: lib/OpenQA/Worker/Common.pm [hunk 3]
```diff
@@ -336,31 +340,35 @@ sub call_websocket {
...
+                log_debug("## worker_status timer will be sending status each $worker_status_timer seconds")
+                  if $verbose;
```
Comment (okurz) [inline lib/OpenQA/Worker/Common.pm]: I don't undersand what thas flag is for. Isn't log_debug implicit for "verbose"?

File: lib/OpenQA/Worker/Commands.pm (hunk 1 above)
Comment (mudler) [inline lib/OpenQA/Worker/Commands.pm]: Because that message is sent right away when the job starts.

File: lib/OpenQA/Worker/Common.pm (hunk 2 above)
Comment (mudler) [inline lib/OpenQA/Worker/Common.pm]: those are temporary variables already

File: lib/OpenQA/Worker/Common.pm (hunk 3 above)
Comment (mudler) [inline lib/OpenQA/Worker/Common.pm]: it's the option flag --verbose in the command line.

**Comment by mudler** [comment]
@okurz it's a wip, and was discussed already with @coolo to remove (now) duplicated workload on the websocket server.

File: lib/OpenQA/Worker/Common.pm (hunk 2 above)
Comment (okurz) [inline lib/OpenQA/Worker/Common.pm]: I meant

```perl
                my $foo = ($i - 1) * (DEAD_DETECTION_THRESHOLD - MIN_STATUS_TIMER);
                my $worker_status_timer = int(MIN_STATUS_TIMER + ( $foo / ($imax - 1)));
```

Comment (mudler) [inline lib/OpenQA/Worker/Common.pm]: Right, and that would add to the code *just another* useless variable for **1** statement forcing even the logic to come up with a variable name, that would be *meaningless* anyway:
```perl
my $foo = ($i - 1) * (DEAD_DETECTION_THRESHOLD - MIN_STATUS_TIMER);
my $bar = ( $foo / ($imax - 1));
my $worker_status_timer = int(MIN_STATUS_TIMER + $bar);
```
How you are going to call `$foo` and `$bar` then? I don't see a good reason to do that
At this point, if the line break is the problem, it's better to just shrink the constant name, e.g:
```perl
my $status_timer = int(MIN_TIMER + ((($i - 1) * (MAX_TIMER - MIN_TIMER)) / ($imax - 1)));
```

**Comment by mudler** [comment]
Closing, superseded by #1486

**Pull request closed without merging**